crates/cheatcodes/src/fs.rs

```diff
@@ -626,6 +630,224 @@ fn prompt(
     }
 }
 
+impl Cheatcode for getBroadcastCall {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { contractName, chainId, txType } = self;
+
+        let reader = BroadcastReader::new(contractName.clone(), *chainId, &state.config.root)?
+            .with_tx_type(*txType);
+
+        let broadcast = reader.read_latest()?;
+
+        let results = reader.search_broadcast(broadcast)?;
+
+        let summaries = reader.parse_results(results);
+
+        if let Some(summary) = summaries.first() {
+            return Ok(summary.abi_encode());
+        }
+
+        Ok(Default::default())
+    }
+}
+
+impl Cheatcode for getBroadcasts_0Call {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { contractName, chainId, txType } = self;
+
+        let reader = BroadcastReader::new(contractName.clone(), *chainId, &state.config.root)?
+            .with_tx_type(*txType);
+
+        let broadcasts = reader.read_all()?;
+
+        let mut summaries = broadcasts
+            .into_iter()
+            .flat_map(|broadcast| -> Result<Vec<BroadcastTxSummary>> {
+                let results = reader.search_broadcast(broadcast)?;
+                Ok(reader.parse_results(results))
+            })
+            .flatten()
+            .collect::<Vec<BroadcastTxSummary>>();
+
+        // Sort by descending block number
+        summaries.sort_by(|a, b| b.blockNumber.cmp(&a.blockNumber));
+
+        Ok(summaries.abi_encode())
+    }
+}
+
+impl Cheatcode for getBroadcasts_1Call {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { contractName, chainId } = self;
+
+        let reader = BroadcastReader::new(contractName.clone(), *chainId, &state.config.root)?;
+
+        let broadcasts = reader.read_all()?;
+
+        let mut summaries = broadcasts
+            .into_iter()
+            .flat_map(|broadcast| -> Result<Vec<BroadcastTxSummary>> {
+                let results = reader.search_broadcast(broadcast)?;
+                Ok(reader.parse_results(results))
+            })
+            .flatten()
+            .collect::<Vec<BroadcastTxSummary>>();
+
+        // Sort by descending block number
+        summaries.sort_by(|a, b| b.blockNumber.cmp(&a.blockNumber));
+
+        Ok(summaries.abi_encode())
+    }
+}
+
+struct BroadcastReader {
```

Comment:
I think we should add some comments to struct and functions. Also, would it make sense to move the reader in a common lib so it could also be used from other places, wdyt? (like if we want to provide a cli for broadcast history)

Reply:
This would cause cyclic deps between common and script-seqeunce. I could move it to script-sequence? @grandizzy

Reply:
ah, I see, I think should work. @klkvr wdyt, would script-sequence be a good place or not worth moving it?